panic: runtime error: invalid memory address or nil pointer dereference

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### Nuclei version:
<!-- You can find current version of nuclei with "nuclei -version" --> Current Version: 2.8.3
<!-- We only accept issues that are reproducible on the latest version of nuclei. -->
<!-- You can find the latest version of project at https://github.com/projectdiscovery/nuclei/releases/ -->

### Current Behavior:

```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xf637da]

goroutine 136 [running]:
net/url.(*URL).Hostname(...)
        /usr/lib/go-1.19/src/net/url/url.go:1144
github.com/projectdiscovery/nuclei/v2/pkg/input.(*Helper).convertInputToType(0xc000510fa0, {0xc008972060, 0x2c}, 0x1, {0x0, 0x0})
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.8.3/pkg/input/input.go:97 +0x21a
github.com/projectdiscovery/nuclei/v2/pkg/input.(*Helper).Transform(0x0?, {0xc008972060?, 0x0?}, 0x0?)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.8.3/pkg/input/input.go:39 +0xca
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer.(*Executer).Execute(0xc0031efc60, 0xc0088ddba0)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.8.3/pkg/protocols/common/executer/executer.go:75 +0x2c6
github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeModelWithInput.func2.1(0x0?, 0x0?, 0xc0088ddb00)
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.8.3/pkg/core/execute.go:146 +0x182
created by github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeModelWithInput.func2
        /home/kali/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.8.3/pkg/core/execute.go:131 +0x745
```

### Expected Behavior:
<!-- A concise description of what you expected to happen. --> Nuclei should show the findings

### Steps To Reproduce:
<!--
Example: steps to reproduce the behavior:
1. running Nuclei with default paramater - > cat live.txt | nuclei -o find.txt


### Anything else:
<!-- Links? References? Screenshots? Anything th
![error](https://user-images.githubusercontent.com/13837228/209459690-165af72e-39c2-4435-a200-066b64f853dd.png)
at will give us more context about the issue that you are encountering! -->

--- a/v2/pkg/input/input.go
+++ b/v2/pkg/input/input.go
@@ -100,7 +100,7 @@ func (h *Helper) convertInputToType(input string, inputType inputType, defaultPo
 		if hasHost {
 			return host
 		}
-		if isURL {
+		if isURL && uri != nil {
 			return uri.Hostname()
 		}
 		return input

